### PR TITLE
Provide option to disable kube-proxy

### PR DESCRIPTION
**Pull request by schinmai-akamai**
# Description

RKE2 has changed the method to disable kube-proxy. It is no longer sufficient to list rke2-kube-proxy in the rke_disable list. Instead, kube-proxy must be explicitly disabled with a special config flag.

References: 
1. https://docs.rke2.io/advanced?_highlight=disable#disabling-server-charts
2. https://github.com/rancher/rke2/issues/2728

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Small minor change not affecting the Ansible Role code (Github Actions Workflow, Documentation etc.)

## How Has This Been Tested?
Deployed a cluster with `disable_kube_proxy` set to `true` - the kube-proxy pods no longer come up in `kube-system` namespace.

### CODE REVIEW

**Comment by schinmai-akamai** [comment]
@MonolithProjects can I get a quick review on this when you get a chance?

File: templates/config.yaml.j2 [hunk 1]
```diff
@@ -36,6 +36,9 @@ node-label:
 {% endif %}
 snapshotter: {{ rke2_snapshooter }}
 node-name: {{ inventory_hostname }}
+{% if ( disable_kube_proxy is defined) %}
```
Comment (MonolithProjects) [inline templates/config.yaml.j2:39]: I think it would be better to evaluate if the variable is `true` or `false`. Maybe you could use `sameas` for this.

**Review by MonolithProjects — CHANGES_REQUESTED** [review]

File: templates/config.yaml.j2 (hunk 1 above)
Comment (MonolithProjects) [reply templates/config.yaml.j2:39]: Sure, that should be fine too.

Comment (schinmai-akamai) [reply templates/config.yaml.j2]: done

**Comment by schinmai-akamai** [comment]
@MonolithProjects the tests seem to be failing with a weird error.. can you help take a look?

**Comment by schinmai-akamai** [comment]
Yes, I've tested this locally, and things look good.

**Review by MonolithProjects — APPROVED** [review]
LGTM

**Pull request merged**